core/src/test/java/org/infinispan/persistence/WriteBehindFaultToleranceTest.java

```diff
@@ -49,43 +51,42 @@
    @Test
    public void testBlockingOnStoreAvailabilityChange() {
       Cache<Object, Object> cache = createManagerAndGetCache(false, 1);
+      PollingPersistenceManager pm = new PollingPersistenceManager();
+      TestingUtil.replaceComponent(cache, PersistenceManager.class, pm, true);
       AdvancedAsyncCacheWriter asyncWriter = TestingUtil.getFirstWriter(cache);
       DummyInMemoryStore store = (DummyInMemoryStore) TestingUtil.extractField(AdvancedAsyncCacheWriter.class, asyncWriter, "actual");
       store.setAvailable(true);
       cache.put(1, 1);
       eventually(() -> store.load(1) != null);
       assertEquals(1, store.size());
 
-      PersistenceManager pm = TestingUtil.extractComponent(cache, PersistenceManager.class);
       store.setAvailable(false);
       assertFalse(store.isAvailable());
 
-      TestingUtil.sleepThread(AVAILABILITY_INTERVAL * 5);
+      int pollCount = pm.pollCount.get();
       // PM & AsyncWriter should still be available as the async modification queue is not full
       assertTrue(asyncWriter.isAvailable());
       assertFalse(TestingUtil.extractField(asyncWriter, "delegateAvailable"));
+      // Wait until the stores availability has been checked before asserting that the pm is still available
+      eventually(() -> pm.pollCount.get() > pollCount);
       assertTrue(pm.isAvailable());
 
       // Add entries >= modification queue size so that store is no longer available
       cache.putAll(intMap(0, 10));
       assertEquals(1, store.size());
 
-      TestingUtil.sleepThread(AVAILABILITY_INTERVAL * 5);
       // PM and writer should not be available as the async modification queue is now oversubscribed and the delegate is still unavailable
       assertFalse(asyncWriter.isAvailable());
-      assertFalse(pm.isAvailable());
+      eventually(() -> !pm.isAvailable());
```

Comment:
Sorry, but I don't think this is right: IMO the new `eventually` calls should be exactly where the `sleepThread` calls were

Reply:
done

Reply:
@ryanemerson I should have been more clear: I was thinking about all the new `eventually` calls (except for `eventually(() -> 10 == store.size());`, that one's fine where it is) :)

Reply:
@danberindei I think that's what I have done, except one of the `Thread.sleep` calls was in fact redundant so I don't need to replace with an eventually call.

Reply:
Ok, when I looked it seemed like you only changed this instance. Merging...